packages/block-editor/src/components/url-input/index.js

```diff
@@ -18,233 +18,286 @@ import {
+		</BaseControl>
+	);
+}
 
-		this.isUpdatingSuggestions = false;
+function defaultRenderSuggestions( {
+	suggestions,
+	suggestionsListProps,
+	handleSuggestionClick,
+	buildSuggestionItemProps,
+	className,
+	selectedSuggestion,
+} ) {
+	return (
+		<Popover position="bottom" noArrow focusOnMount={ false }>
+			<div
+				{ ...suggestionsListProps }
+				className={ classnames(
+					'block-editor-url-input__suggestions',
+					`${ className }__suggestions`
+				) }
+			>
+				{ suggestions.map( ( suggestion, index ) => (
+					<Button
+						{ ...buildSuggestionItemProps( suggestion, index ) }
+						key={ suggestion.id }
+						className={ classnames(
+							'block-editor-url-input__suggestion',
+							{
+								'is-selected': index === selectedSuggestion,
+							}
+						) }
+						onClick={ () => handleSuggestionClick( suggestion ) }
+					>
+						{ suggestion.title }
+					</Button>
+				) ) }
+			</div>
+		</Popover>
+	);
+}
 
-		this.state = {
-			suggestions: [],
-			showSuggestions: false,
-			selectedSuggestion: null,
+function URLInput( {
+	value,
+	onChange,
+	label,
+	className,
+	isFullWidth,
+	placeholder = __( 'Paste URL or type to search' ),
+	disableSuggestions,
+	speak,
+	debouncedSpeak,
+	__experimentalRenderControl: renderControl = defaultRenderControl,
+	__experimentalRenderSuggestions: renderSuggestions = defaultRenderSuggestions,
+	__experimentalFetchLinkSuggestions,
+	__experimentalHandleURLSuggestions: handleURLSuggestions = false,
+	__experimentalShowInitialSuggestions: showInitialSuggestions = false,
+	setTimeout,
+	parentAutocompleteRef,
+} ) {
+	const [ isLoading, setLoading ] = useState( false );
+	const [ suggestions, setSuggestions ] = useState( [] );
+	const [ showSuggestions, setShowSuggestions ] = useState( false );
+	const [ selectedSuggestion, setSelectedSuggestion ] = useState( null );
+	const [ prevInstanceId, setPrevInstanceId ] = useState( null );
+	const [ suggestionsListboxId, setSuggestionsListboxId ] = useState( '' );
+	const [ suggestionOptionIdPrefix, setSuggestionOptionIdPrefix ] = useState(
+		''
+	);
+	const [ prevDisableSuggestions, setPrevDisableSuggestions ] = useState(
+		null
+	);
+	const [ prevShowInitalSuggestions, setShowInitialSuggestions ] = useState(
+		null
+	);
+
+	const autocompleteRef = useRef( parentAutocompleteRef );
+	const inputRef = useRef( null );
+	const pendingRequest = useRef( null );
+	const scrollingIntoView = useRef( false );
+	const suggestionNodes = useRef( [] );
+	const instanceId = useInstanceId( URLInput );
+
+	const {
+		fetchLinkSuggestions = __experimentalFetchLinkSuggestions,
+	} = useSelect( ( select ) => {
+		if ( typeof __experimentalFetchLinkSuggestions === 'function' ) {
+			return {};
+		}
 
-			suggestionsListboxId: '',
-			suggestionOptionIdPrefix: '',
+		const { getSettings } = select( 'core/block-editor' );
+		const settings = getSettings();
+		return {
+			fetchLinkSuggestions: settings.__experimentalFetchLinkSuggestions,
 		};
-	}
+	} );
```

Comment:
You need add `__experimentalFetchLinkSuggestions` as dependency
```suggestion
	}, [ __experimentalFetchLinkSuggestions ] );
```